src/test/java/ru/yandex/clickhouse/integration/RowBinaryStreamTest.java

```diff
@@ -150,6 +206,77 @@ public void writeTo(ClickHouseRowBinaryStream stream) throws IOException {
         Assert.assertEquals(rs.getDouble("float32"), 123.456);
         Assert.assertEquals(rs.getDouble("float64"), 42.21);
 
+        final Date[] dateArray = (Date[]) rs.getArray("dateArray").getArray();
+        Assert.assertEquals(dateArray.length, dates1.length);
+        for (int i = 0; i < dateArray.length; i++) {
+            // expected is Date at start of the day in server timezone
+            DateTime dt = new DateTime(dates1[i].getTime())
```

Comment:
Your suggestion looks right. Wouldn't it be neccessarry to check wih clickhouse-server developers?